homeassistant/components/vizio/config_flow.py

```diff
@@ -97,16 +97,51 @@ def __init__(self, config_entry: ConfigEntry) -> None:
     ) -> Dict[str, Any]:
         """Manage the vizio options."""
         if user_input is not None:
+            if user_input.get(CONF_APPS_TO_INCLUDE_OR_EXCLUDE):
+                user_input[CONF_APPS] = {
+                    user_input[CONF_INCLUDE_OR_EXCLUDE]
+                    .lower(): user_input[CONF_APPS_TO_INCLUDE_OR_EXCLUDE]
+                    .copy()
+                }
+
+                user_input.pop(CONF_INCLUDE_OR_EXCLUDE)
+                user_input.pop(CONF_APPS_TO_INCLUDE_OR_EXCLUDE)
+
             return self.async_create_entry(title="", data=user_input)
 
-        options = {
-            vol.Optional(
-                CONF_VOLUME_STEP,
-                default=self.config_entry.options.get(
-                    CONF_VOLUME_STEP, DEFAULT_VOLUME_STEP
-                ),
-            ): vol.All(vol.Coerce(int), vol.Range(min=1, max=10))
-        }
+        if self.config_entry.data[CONF_DEVICE_CLASS] == DEVICE_CLASS_TV:
+            default_include_or_exclude = (
+                CONF_EXCLUDE
+                if self.config_entry.options
+                and CONF_EXCLUDE in self.config_entry.options.get(CONF_APPS)
+                else CONF_EXCLUDE
+            )
+            options = {
+                vol.Optional(
+                    CONF_VOLUME_STEP,
+                    default=self.config_entry.options.get(
+                        CONF_VOLUME_STEP, DEFAULT_VOLUME_STEP
+                    ),
+                ): vol.All(vol.Coerce(int), vol.Range(min=1, max=10)),
+                vol.Optional(
+                    CONF_INCLUDE_OR_EXCLUDE, default=default_include_or_exclude.title(),
+                ): vol.In([CONF_INCLUDE.title(), CONF_EXCLUDE.title()]),
+                vol.Optional(
+                    CONF_APPS_TO_INCLUDE_OR_EXCLUDE,
+                    default=self.config_entry.options.get(CONF_APPS, {}).get(
+                        default_include_or_exclude, []
+                    ),
+                ): cv.multi_select(VizioAsync.get_apps_list()),
+            }
+        else:
+            options = {
+                vol.Optional(
+                    CONF_VOLUME_STEP,
```

Comment:
Since this option is the same for both device types, we can set this option first before checking device type.